Sources/RemoteNotification.swift

```diff
@@ -33,28 +33,39 @@ public struct RemoteNotification: CustomStringConvertible, Equatable {
     /// The identifier for the UIUserNotificationCategory associated with the push notification. Derived from userInfo["category"]
     public let categoryIdentifier: String?
     /// Custom fields on the remote notification.
-    public let userInfo: [String : AnyObject]
+    public let userInfo: [String : Any]
     /// The dictionary representation of the remote notification.
-    public let remoteNotificationDictionary: [String : AnyObject]
+    public let remoteNotificationDictionary: [String : Any]
     
-    public init?(remoteNotification: [NSObject : AnyObject]?) {
-        guard let remoteNotification = remoteNotification as? [String : AnyObject] else {
+    // MARK: Equatable
+    
+    public static func ==(lhs: RemoteNotification, rhs: RemoteNotification) -> Bool {
+        return lhs.alert == rhs.alert
+            && lhs.badge == rhs.badge
+            && lhs.sound == rhs.sound
+            && lhs.contentAvailable == rhs.contentAvailable
+            && lhs.categoryIdentifier == rhs.categoryIdentifier
+            && NSDictionary(dictionary: lhs.userInfo) == NSDictionary(dictionary: rhs.userInfo)
+    }
+    
+    public init?(remoteNotification: [AnyHashable : Any]?) {
+        guard let remoteNotification = remoteNotification as! [String : Any]? else {
             return nil
         }
         
         // Parse the notification payload.
         // See https://developer.apple.com/library/ios/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/Chapters/TheNotificationPayload.html for more information on expected values.
         alert = Alert(remoteNotification: remoteNotification)
         
-        let apnsDictionary = remoteNotification[APSServiceKey] as? [String : AnyObject]
-        badge = apnsDictionary?[badgeKey] as? Int
-        sound = apnsDictionary?[soundKey] as? String
+        let apnsDictionary = remoteNotification[apsServiceKey] as! [String : Any]?
```

Comment:
I can understand the use of `!` if API from Apple is promising it. However, SuperDelegate is providing itself as a robust wrapper around potential idiosyncrasies between API versions. My recommendation would be to avoid `!`, and throw a `APIContractViolationError` or equivalent when you get something from Cocoa that you didn't expect. This could help preserve a very robust SuperDelegate moving forward.